### ISSUE TITLE
InputText is written wrongly

### ISSUE THREAD

**Issue by smsimone** [mobile-dev-inc/maestro#107]
As stated in the screenshot below, instead of writing `luigiverdi@toduba.it`, the driver does not write the @ symbol but adds the quotes instead

![Schermata 2022-09-01 alle 16 13 25](https://user-images.githubusercontent.com/64575218/187935779-6a18d597-903e-4ed9-86ac-1c2d5bc4c22a.png)

**Comment by dmitry-zaitsev** [comment]
I suspect this might be related to the locale set on your device. Could you try setting it to English / US to see if that solves the problem?

<img width="365" alt="image" src="https://user-images.githubusercontent.com/2990722/187938765-be9902b2-04f9-41c4-8a89-f1ad5188c9b1.png">

**Comment by smsimone** [comment]
I tried with both English (USA) and English (UK), but it still writes the quotes instead of @

**Comment by dmitry-zaitsev** [comment]
What is a locale that is set on your Mac? Also, what is the language of an input method?

I am wondering if that could have an effect.

**Comment by smsimone** [comment]
My Mac's locale is Italian, on the emulator I've tried English and Italian keyboard but in both cases it still writes " instead of @.

I've noticed that by running the exact same flow on an Android emulator, the virtual keyboard is shown and the drivers simulates the taps, while on iOS the virtual keyboard is not shown. Is it an intended behaviour?

**Comment by dmitry-zaitsev** [comment]
Yes, we interact with the device via Facebook's IDB that in turn talks directly to Apple frameworks.

Thanks for the context, I was able to reproduce the issue 👍

**Comment by MenilV** [comment]
without actually opening a different issue, I've noticed maestro has issues properly parsing multi-worded input on `inputText`. I've tried several test cases:

- [x] - inputText: "username@email.com"
- [x] - inputText: username@email.com
- [ ] - inputText: username@email.com and something
- [ ] - inputText: "username@email.com and something"


All of these were tested on Android with the emulator and the behavior is consistent. also went through the code but it seems as if the YAML parses defaults to `null` when it fails to parse the inputText `String`. thoughts?

**Comment by Holospin** [comment]
> without actually opening a different issue, I've noticed maestro has issues properly parsing multi-worded input on `inputText`. I've tried several test cases:
> 
> * [x]  - inputText: "[username@email.com](mailto:username@email.com)"
> * [x]  - inputText: [username@email.com](mailto:username@email.com)
> * [ ]  - inputText: [username@email.com](mailto:username@email.com) and something
> * [ ]  - inputText: "[username@email.com](mailto:username@email.com) and something"
> 
> All of these were tested on Android with the emulator and the behavior is consistent. also went through the code but it seems as if the YAML parses defaults to `null` when it fails to parse the inputText `String`. thoughts?
> 
> edit: test cases that fail produce an empty string hence the suspicion towards the parses defaulting to null (or empty)

I am also running into the same issue as I was just about to create an issue. Glad to see someone already mentioned it

**Comment by dmitry-zaitsev** [comment]
Hey folks, just wanted to call out that this is a high priority for us, we are going to look into it early next week

**Comment by dmitry-zaitsev** [comment]
@MenilV thanks for reporting and the test cases. The issue is different from what @smsimone mentioned, but is nonetheless valid. I have addressed it in #122 (going to be released today/tomorrow).

@smsimone I have confirmed that the issue you are facing is related to the system input method (aka keyboard language). The simplest way to fix it is to switch to a US English keyboard layout. In the meantime, I am looking for a smarter way to resolve it.
